Fixme: SA1019: grpc.WithDialer is deprecated

Tracking issue for some remaining linting issues to be fixed after https://github.com/moby/moby/pull/39668 is merged.

```
daemon/daemon.go:885:3:                               SA1019: grpc.WithDialer is deprecated: use WithContextDialer instead.  Will be supported throughout 1.x.  (staticcheck)
```

These warnings are currently suppressed in `hack/validate/golangci-lint.yml` (after https://github.com/moby/moby/pull/39668 is merged), so when this is fixed, those lines should be removed

--- a/daemon/daemon.go
+++ b/daemon/daemon.go
@@ -887,7 +887,7 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 	gopts := []grpc.DialOption{
 		grpc.WithInsecure(),
 		grpc.WithBackoffMaxDelay(3 * time.Second),
-		grpc.WithDialer(dialer.Dialer),
+		grpc.WithContextDialer(dialer.ContextDialer),
 
 		// TODO(stevvooe): We may need to allow configuration of this on the client.
 		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(defaults.DefaultMaxRecvMsgSize)),

--- a/hack/validate/golangci-lint.yml
+++ b/hack/validate/golangci-lint.yml
@@ -46,10 +46,6 @@ issues:
     - text: "G202: SQL string concatenation"
       linters:
         - gosec
-    # FIXME temporarily suppress these. See #39928
-    - text: "SA1019: grpc.WithDialer is deprecated"
-      linters:
-        - staticcheck
     # FIXME temporarily suppress these. See #39924
     - text: "SA1019: h.Xattrs is deprecated: Use PAXRecords instead"
       linters:

